pkg/metrics/eventmetrics/eventmetrics.go

```diff
@@ -25,7 +25,7 @@ var (
 		Name:        "events_total",
 		Help:        "The total number of Tetragon events",
 		ConstLabels: nil,
-	}, []string{"type", "namespace", "workload", "pod", "binary"})
+	}, metrics.FilterMetricLabels(append([]string{"type"}, consts.KnownMetricLabelFilters...)...))
```

Comment:
`metrics.FilterMetricLabels` doesn't filter correctly in metrics definitions, because these definitions are evaluated before `option.Config.MetricsLabelFilter` is set.

To support configuring labels dynamically, we need to create Prometheus metrics after the config is evaluated, not as global variables. This can be done by defining an internal metric type wrapping the Prometheus metric type. The internal type would be used in global metrics definitions, but the Prometheus type would be initialized later, when the metric is registered.

I suggest defining such type in `pkg/metrics`:
```
// GranularCounter wraps *prometheus.CounterVec. It's intended to be used to
// define metrics with labels set at register time.
type GranularCounter struct {
	counter *prometheus.CounterVec
	prometheus.CounterOpts
	labels   []string
	register sync.Once
}

// MustNewGranularCounter creates a new GranularCounter. It adds labels defined in
// const.KnownMetricLabelFilters to the labels passed in an argument. The function panics if the
// labels passed contain any of the labels from const.KnownMetricLabelFilters.
func MustNewGranularCounter(opts prometheus.CounterOpts, labels []string) *GranularCounter {
	for _, label := range labels {
		if slices.Contains(consts.KnownMetricLabelFilters, label) {
			panic(fmt.Sprintf("labels passed to GranularCounter can't contain any of the following: %v. These labels are added by Tetragon.", consts.KnownMetricLabelFilters))
		}
	}
	return &GranularCounter{
		CounterOpts: opts,
		labels:      append(labels, consts.KnownMetricLabelFilters...),
	}
}

// ToProm converts GranularCounter to prometheus.CounterVec. The first time it's called, it
// evaluates the labels and creates the underlying value.
func (m *GranularCounter) ToProm() *prometheus.CounterVec {
	m.register.Do(func() {
		m.labels = FilterMetricLabels(m.labels...)
		m.counter = NewCounterVecWithPod(m.CounterOpts, m.labels)
	})
	return m.counter
}
```
the we can define this metric as:
```
EventsProcessed = metrics.MustNewGranularCounter(prometheus.CounterOpts{
    Namespace:   consts.MetricsNamespace,
    Name:        "events_total",
    Help:        "The total number of Tetragon events",
    ConstLabels: nil,
}, []string{"type"})
```
and convert it to the Prometheus type whenever it's used:
```
EventsProcessed.ToProm()
```

Let me know if there are any questions/concerns about this approach.

Reply:
I'll work through implementing this!
Thanks for the suggestion and help troubleshooting!